docs/use-the-network/run-a-network-server/debug-with-sniffer.mdx

```diff
@@ -0,0 +1,280 @@
+---
+id: debug-with-sniffer
+sidebar_label: Debug With Sniffer
+slug: /use-the-network/debug-with-sniffer
+---
+
+# Debug With Sniffer
+
+Confirm router runtime behavior using a virtual LoRaWAN device with a
+software-based protocol sniffer.
+
+## Dependencies
+
+1. Ensure [Router](https://github.com/helium/router.git) is running and
+   [Console](https://github.com/helium/console.git) is connected to it.
+1. The wallet to be used requires enough HNT to burn for joining the virtual
+   device to the network.
+   + It's the same fee as for a physical device.
+   + See
+   [Setup Data Only Hotspot](https://docs.helium.com/mine-hnt/data-only-hotspots/#transactions--cost)
+1. Clone each of these repos:
+    + [virtual-lorawan-device](https://github.com/helium/virtual-lorawan-device)
+    + [lorawan-sniffer](https://github.com/helium/lorawan-sniffer)
+    + [gateway-rs](https://github.com/helium/gateway-rs)
+    + [helium-wallet-rs](https://github.com/helium/helium-wallet-rs)
+
+Those packages require [Rust](https://rust-lang.org/) when building from
+source.
+
+Everything may run locally on a laptop/workstation, or use just the
+Rust-based tools locally for connecting to a remote Router & Console.
+
+## Overview
+
+The approximate sequence:
+
+1. Create a virtual device
+1. Use the wallet to pay for joining a device to the network
+1. Device gets added via Console, which then will be "pending" for about 15 minutes
+1. Start local data-only hotspot
+1. Configure sniffer to use that hotspot
+1. Connect the virtual device to sniffer
+1. Wait for "pending" to resolve on Console
+1. Finally, the virtual device joins the network
+
+Order of operations below is significant.
+
+# Setup
+
+## Compile
+
+Because there are multiple repos each in their own subdirectory, examples
+below assume everything is located beneath the `~/helium/` path.
+
+Ensure everything compiles successfully.
+
```

Comment:
perhaps code block might make this render nicely?

Reply:
Fixed in commit d5bb98b

Nice catch!  I didn't check if there's explicit styling for `bash` blocks that the previous indented block wouldn't get.